sda-commons-server-dropwizard/src/main/java/org/sdase/commons/server/dropwizard/bundles/configuration/generic/GenericLookupYamlConfigurationFactory.java

```diff
@@ -72,6 +78,7 @@ protected T build(JsonNode node, String path) throws IOException, ConfigurationE
         .map(m -> m.expand(configurationRuntimeContext.getDefinedKeys()))
         .flatMap(List::stream)
         .sorted()
+        .filter(m -> !lookupTracker.lookedUpKeys().contains(m.getContextKey()))
```

Comment:
`lookedUpKeys()` will create a new unmodifiable `Set` for every invocation (see `SystemPropertyAndEnvironmentLookup`). The `LookupTracker` could expose a `contains` method that returns `true` or `false`.

Reply:
If you are afraid of the performance, what about [this solution](https://github.com/SDA-SE/sda-dropwizard-commons/compare/0e35106bf841353b38670b3a87f2b6f68bcd2c75..9946c82cd462e5d78c328ceacf713038779914b5)?